### PR TITLE
feat(RELEASE-996): availability metrics for release-service

### PR DESCRIPTION
**Pull request by theflockers**
This commit adds availability record rules and tests for the release-service availability metrics.

### CODE REVIEW

**Comment by amisstea** [comment]
@theflockers the EC failures should be resolved by rebasing your branch against `main`.

File: rhobs/recording/release_service_availability_recording_rules.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+spec:
+
+  groups:
+    - name: release_service_general_availability
+      interval: 1m
+      rules:
+        - record: konflux_up
+          expr: |
+            label_replace(
+              avg(release_service_check_gauge) without(Name) OR
```
Comment (yftacherzog) [inline rhobs/recording/release_service_availability_recording_rules.yaml:16]: As we mention the label `Name` here, I think we should also have it inside the input series to verify things are aggregated as expected.

Comment (theflockers) [reply rhobs/recording/release_service_availability_recording_rules.yaml:16]: True. As I am artificially adding the "release"  one, it duplicates the Name - hence I had to remove it, before re-adding it. But I forgot to update with the last operation. Fixed!

**Comment by theflockers** [comment]
The PR got weird after the rebase. I will close and recreate.

**Comment by theflockers** [comment]
> The PR got weird after the rebase. I will close and recreate.

could fix it!!

**Comment by yftacherzog** [comment]
> > The PR got weird after the rebase. I will close and recreate.
> 
> Less bad. I could fix it!!

I can't see the change you mentioned though :)

**Comment by theflockers** [comment]
> > > The PR got weird after the rebase. I will close and recreate.
> > 
> > 
> > Less bad. I could fix it!!
> 
> I can't see the change you mentioned though :)

Yeah. I had to force push it: https://github.com/redhat-appstudio/o11y/blob/9ee0c5419b8bd63ae54333fb6134bd5dca7558aa/rhobs/recording/release_service_availability_recording_rules.yaml#L20

Anyhow, something else is breaking now. Checking it.

File: rhobs/recording/release_service_availability_recording_rules.yaml (hunk 1 above)
Comment (yftacherzog) [reply rhobs/recording/release_service_availability_recording_rules.yaml:16]: The test now shows that it's being added. I think we should also have it in some of the inputs to test how we handle also cases in which it's present in the first place in `release_service_check_gauge`.

**Review by yftacherzog — APPROVED** [review]
lgtm

**Comment by theflockers** [comment]
@yftacherzog please merge this one whenever you can. Thanks!

**Pull request merged**